engine/transaction/record/record_outline.go

```diff
@@ -0,0 +1,126 @@
+package record
+
+import (
+	"context"
+
+	"github.com/bitcoin-sv/go-sdk/spv"
+	trx "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/conv"
+	"github.com/bitcoin-sv/spv-wallet/engine/database"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/engine/transaction"
+	txerrors "github.com/bitcoin-sv/spv-wallet/engine/transaction/errors"
+	"github.com/bitcoin-sv/spv-wallet/engine/transaction/outlines"
+	"github.com/bitcoin-sv/spv-wallet/models/bsv"
+	"github.com/bitcoin-sv/spv-wallet/models/transaction/bucket"
+)
+
+// RecordTransactionOutline will validate, broadcast and save a transaction outline
+func (s *Service) RecordTransactionOutline(ctx context.Context, outline *outlines.Transaction) error {
+	tx, err := trx.NewTransactionFromBEEFHex(outline.BEEF)
+	if err != nil {
+		return txerrors.ErrTxValidation.Wrap(err)
+	}
+
+	if ok, err := spv.VerifyScripts(tx); err != nil {
+		return txerrors.ErrTxValidation.Wrap(err)
+	} else if !ok {
+		return txerrors.ErrTxValidation
+	}
+
+	utxos, err := s.getTrackedUTXOsFromInputs(ctx, tx)
+	if err != nil {
+		return err
+	}
+
+	txID := tx.TxID().String()
+	for _, utxo := range utxos {
+		utxo.Spend(txID)
```

Comment:
just thinking out loud:
From the big picture perspective shouldn't they be spend after broadcast is done?

Reply:
It is as you say - because the database-update (`SaveTX`) it **after** the broadcast.
So my concept is as follows:
- prepare everything, broadcast, if ok - save.
- Additionally, we should consider some redis-locking (but more brainstorming is needed)

Reply:
I understand your approach Krzysiek, but I don't know if from a logical point of view I wouldn't be in favor of moving the loop from utxo.Spend below broadcasting if no error occurred.
It seems to me that it would “read” better, and from a behavioral point of view it probably doesn't matter much, because the changes will be applied in s.repo.SaveTX anyway?

Reply:
I can do that 😉